website/docs/cloud-docs/api-docs/workspaces.mdx

```diff
@@ -2893,6 +2893,183 @@ No response body.
+| [404][] | [JSON API error object][]                       | Workspace not found, data retention policy does not exist, or user unauthorized to perform action |
+
+### Request Body
+
+No request body.
+
+### Sample Request
+
+```shell
+$ curl \
+  --header "Authorization: Bearer $TOKEN" \
+  --header "Content-Type: application/vnd.api+json" \
+  https://app.terraform.io/api/v2/workspaces/workspace-2/relationships/data-retention-policy
+```
+
+### Sample Response
+
+```json
+{
+  "data": {
+    "id": "drp-Kh74zpKVGNWbpugQ",
+    "type": "data-retention-policy-delete-olders",
+    "attributes": {
+      "delete-older-than-n-days": 60
+    },
+    "relationships": {
+      "target": {
+        "data": {
+          "id": "workspace-2",
+          "type": "workspaces"
+        }
+      }
+    }
+  }
+}
+```
+
+## Create or update data retention policy
+
+<EnterpriseAlert>
+This endpoint is exclusive to Terraform Enterprise, and not available in Terraform Cloud.
+</EnterpriseAlert>
+
+`POST /workspaces/:workspace_id/relationships/data-retention-policy`
+
+This endpoint creates a data retention policy for a workspace, or updates the existing policy. Read more about [workspace data retention policies](/terraform/cloud-docs/workspaces/settings/deletion#data-retention-policies).
+
+The endpoint can also be used to change the type of the data retention policy on the workspace. Making a request with the new type of policy will automatically create a data retention policy of the new type and replace the existing workspace policy. For more information on the types of data retention policies, refer to [Data Retention Policy Types](/terraform/cloud-docs/api-docs/data-retention-policies#types-of-data-retention-policies).
+
+| Parameter       | Description                                                                                                                                                         |
+| --------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `:workspace_id` | The workspace ID to update the data retention policy for. Obtain this from the [workspace settings](/terraform/cloud-docs/workspaces/settings) or the [Show Workspace](#show-workspace) endpoint. |
+
+| Status  | Response                  | Reason(s)                                                   |
+| ------- | ------------------------- | ----------------------------------------------------------- |
+| [204][] | No Content                | Successfully updated the workspace data retention policy.   |
+| [404][] | [JSON API error object][] | Workspace not found, or user unauthorized to perform action |
```

Comment:
Non-blocking Nitpick (same as above!): Should we add a period here for consistency?:
```suggestion
| [404][] | [JSON API error object][] | Workspace not found, or user unauthorized to perform action. |
```